src/redux/validation/validation.listeners.tsx

```diff
@@ -0,0 +1,133 @@
+import {currentConfigSelector} from '@redux/selectors';
+import {clusterResourceMapSelector} from '@redux/selectors/resourceMapSelectors';
+import {loadClusterResources, reloadClusterResources} from '@redux/thunks/cluster';
+import {downloadK8sSchema} from '@redux/thunks/downloadK8sSchema';
+import {setRootFolder} from '@redux/thunks/setRootFolder';
+
+import {doesSchemaExist} from '@utils/index';
+
+import {activeProjectSelector, kubeConfigContextSelector} from '@shared/utils/selectors';
+
+import {setConfigK8sSchemaVersion, toggleOPARules, toggleValidation} from './validation.slice';
+import {loadValidation, validateResources} from './validation.thunks';
+
+const loadListener: AppListenerFn = listen => {
+  listen({
+    matcher: isAnyOf(
+      loadClusterResources.fulfilled,
+      reloadClusterResources.fulfilled,
+      setRootFolder.fulfilled,
+      setConfigK8sSchemaVersion,
+      toggleOPARules,
+      toggleValidation
+    ),
+    async effect(_, {dispatch, delay, signal, cancelActiveListeners}) {
+      cancelActiveListeners();
+      await delay(1);
+      const loading = dispatch(loadValidation());
+      signal.addEventListener('abort', () => loading.abort());
+      await loading;
+    },
+  });
+};
+
+const validateListener: AppListenerFn = listen => {
+  listen({
+    matcher: isAnyOf(loadValidation.fulfilled),
+    async effect(_action, {dispatch, getState, cancelActiveListeners, signal, delay}) {
+      cancelActiveListeners();
+
+      // if (getState().resources.previewLoading) {
+      //   return;
+      // }
+
+      const validatorsLoading = getState().validation.status === 'loading';
+      if (validatorsLoading) return;
+
+      await delay(1);
+      if (signal.aborted) return;
+      const response = dispatch(validateResources());
+      signal.addEventListener('abort', () => response.abort());
+      await response;
+    },
+  });
+};
+
+const clusterK8sSchemaVersionListener: AppListenerFn = listen => {
+  listen({
+    matcher: isAnyOf(loadClusterResources.fulfilled),
+    async effect(_, {dispatch, getState, cancelActiveListeners, delay}) {
+      cancelActiveListeners();
+      await delay(1);
+
+      const state = getState();
+
+      const activeProject = activeProjectSelector(state);
+      const clusterResourceMap = clusterResourceMapSelector(state);
+      const currentContext = kubeConfigContextSelector(state);
+      const localSchemaVersion = currentConfigSelector(getState()).k8sVersion;
+      const userDataDir = String(state.config.userDataDir);
+
+      const nodeResources = Object.values(clusterResourceMap).filter(
+        resource => resource.object.apiVersion === 'v1' && resource.kind === 'Node'
+      );
+
+      if (isEmpty(nodeResources)) return;
+
+      const clusterSchemaVersions = [
+        ...new Set(
+          nodeResources.map(resource =>
+            resource.object?.status?.nodeInfo?.kubeletVersion?.split('+')[0].substring(1).trim()
+          )
+        ),
```

Comment:
I would suggest doing the following to avoid error like "cannot call .split() on undefined".
I'm also not sure if the kubeletVersion is in the same object path on all k8s schema versions, but we could leave this solution for now
```suggestion
        ...new Set(
          nodeResources.map(resource => {
            const kubeletVersion = esource.object?.status?.nodeInfo?.kubeletVersion;
            if (typeof kubeletVersion !== 'string') {
              return undefined;
            }
            return kubeletVersion.split('+')[0].substring(1).trim()
          }).filter(isDefined)
        ),
```